### PR TITLE
Added in build version based on git hash

### PR DESCRIPTION
**Pull request by Luminarys**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -5,6 +5,20 @@ set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "bin/")
...
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${PROJECT_SOURCE_DIR}/CMake)
 
+if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/.git")
+  execute_process(
+    COMMAND git log -1 --format=%h
+    OUTPUT_VARIABLE GIT_COMMIT_HASH
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+  )
+else(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/.git")
+  set(GIT_COMMIT_HASH "Not Found")
```
Comment (minus7) [inline CMakeLists.txt]: Why not just let it unset?

File: CMakeLists.txt [hunk 2]
```diff
@@ -5,6 +5,20 @@ set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "bin/")
 add_definitions("-Wall -Wextra -Wno-unused-parameter")
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${PROJECT_SOURCE_DIR}/CMake)
 
+if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/.git")
+  execute_process(
+    COMMAND git log -1 --format=%h
```
Comment (minus7) [inline CMakeLists.txt]: `git describe --always` gives the same output, `git describe --tags` might give a nicer output.

**Comment by SirCmpwn** [comment]
I agree with minus on both counts.

**Comment by Luminarys** [comment]
Alright, should be good.

**Pull request merged**